crates/pallet-domains/src/lib.rs

```diff
@@ -105,6 +105,7 @@ pub(crate) struct ElectionVerificationParams<Balance> {
 
 pub type DomainBlockNumberFor<T> = <<T as Config>::DomainHeader as Header>::Number;
 pub type DomainHashingFor<T> = <<T as Config>::DomainHeader as Header>::Hashing;
+pub type ReceiptHashFor<T> = <<T as Config>::DomainHeader as Header>::Hash;
```

Comment:
I added `ReceiptHashFor` for redability purposes since its just an alias to DomainBlock::Hash and confusing to look at. 

As for the Hashing, we are always using `DomainBlock::Hashing` and but sometimes we were using Blake2b directly giving us H256  and it was awkward to see so many `Into<H256>` and `From<H256>` trait bounds with implicit assumptions.

I do not have a strong opinion on non-generic ones but my understanding was that it would be ideal to use the same hash function unless there is a strong reason to pick another. But I could be wrong here 🤷🏼‍♂️